lib/index.js

```diff
@@ -411,9 +382,68 @@ var StateMachine = stampit({
       this.addState(event.to);
     }, this);
 
+    this.mixin = _(this.events)
+      .mapValues(function (event, name) {
+        return this.buildEvent(name);
+      }, this)
+      .assign({
+        can: this.can.bind(this),
+        cannot: this.cannot.bind(this),
+        is: this.is.bind(this),
+        hasState: this.hasState.bind(this),
+        isFinal: this.isFinal.bind(this)
+      })
+      .value();
+
     this.current = this.initial;
-    return this.initTarget(_.first(context.args));
   }
 });
 
-module.exports = StateMachine;
+var Target = stampit({
+  props: {
+    events: {},
+    callbacks: {},
+    initial: 'none',
+    final: null
+  },
+  static: {
+    StateMachine: StateMachine,
+    Promise: global.Promise || require('es6-promise').Promise,
+    FsmError: FsmError
+  },
+  init: function init (context) {
+    var target = _.first(context.args) || this;
+
+    _.defaults(target, {
```

Comment:
the defaults are defined in `props`, but if `target` is not `this`, then they aren't going to be used, which is why we have to do it twice.  Unless there's a really good reason to support the second parameter, maybe we should drop that functionality, because it's a bit cumbersome.  In other words--we can now mix any object into a state machine *in addition to* mixing a state machine into any object, but do we need both?